pkgs/by-name/re/resticprofile/package.nix

```diff
@@ -0,0 +1,65 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+}:
+
+buildGoModule rec {
+  pname = "resticprofile";
+  version = "0.26.0";
+
+  src = fetchFromGitHub {
+    owner = "creativeprojects";
+    repo = "resticprofile";
+    rev = "v${version}";
+    hash = "sha256-y2rBUivs5yxlD/0uFwXW/Zc+o3foTmydwtEkyiuJwyw=";
+  };
+
+  vendorHash = "sha256-Qi7uhMXaWqI4NmYi+XTR15SyiUGhRiXPZmVud6aTM4s=";
+
+  ldflags = [
+    "-X main.commit=${src.rev}"
+    "-X main.date=unknown"
+    "-X main.builtBy=nixpkgs"
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  preCheck = ''
+    rm battery_test.go # fetches battery data
+    rm update_test.go # uses network
+    rm config/path_test.go # gets confused by build env
+    rm lock/lock_test.go # needs ping
+    rm preventsleep/caffeinate_test.go # communicates with dbus
+    rm priority/ioprio_test.go # IDK
+    rm restic/downloader_test.go # uses network
+    rm schedule/schedule_test.go # uses systemctl
+    rm util/tempdir_test.go # gets confused by build env
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 $GOPATH/bin/resticprofile -t $out/bin
+
+    installShellCompletion --cmd resticprofile \
+        --bash <($out/bin/resticprofile generate --bash-completion) \
+        --zsh <($out/bin/resticprofile generate --zsh-completion)
+
+    runHook postInstall
+  '';
+
+  meta = {
+    changelog = "https://github.com/creativeprojects/resticprofile/releases/tag/${src.rev}";
+    description = "Configuration profiles manager for restic backup";
+    homepage = "https://creativeprojects.github.io/resticprofile/";
+    license = with lib.licenses; [
+      gpl3Only
+      lgpl3 # bash shell completion
+    ];
+    mainProgram = "resticprofile";
+    maintainers = with lib.maintainers; [ tomasajt ];
+    platforms = lib.platforms.all;
```

Comment:
This should not be `all`, as that includes really strange platforms like wasm and m68k. You should make this more specific.